src/Documentation/Advanced-Concepts/Docker-Deployment.md

```diff
@@ -0,0 +1,617 @@
+                    siloHost.Dispose();
+                    Console.WriteLine($"Orleans silo '{siloHost.Name}' shutdown.");
+                }
+                catch (Exception exc)
+                {
+                    siloHost.ReportStartupError(exc);
+                    Console.Error.WriteLine(exc);
+                    return 1;
+                }
+            }
+            return 0;
+        }
+    }
+}
+```
+
+`Program.cs` (Silo):
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Orleans.Runtime.Configuration;
+
+namespace OrleansSilo
+{
+    public class Program
+    {
+        private static OrleansHostWrapper hostWrapper;
+
+        static int Main(string[] args)
+        {
+            int exitCode = InitializeOrleans();
+
+            Console.WriteLine("Press Enter to terminate...");
+            Console.ReadLine();
+
+            exitCode += ShutdownSilo();
+
+            return exitCode;
+        }
+
+        private static int InitializeOrleans()
+        {
+            var config = new ClusterConfiguration();
+            config.Globals.DataConnectionString = "[AZURE STORAGE CONNECTION STRING HERE]";
+            config.Globals.DeploymentId = "Orleans-Docker";
+            config.Globals.LivenessType = GlobalConfiguration.LivenessProviderType.AzureTable;
+            config.Globals.ReminderServiceType = GlobalConfiguration.ReminderServiceProviderType.AzureTable;
+            config.Defaults.PropagateActivityId = true;
+            config.Defaults.ProxyGatewayEndpoint = new IPEndPoint(IPAddress.Any, 10400);
+            config.Defaults.Port = 10300;
+            var ips = Dns.GetHostAddressesAsync(Dns.GetHostName()).Result;
+            config.Defaults.HostNameOrIPAddress = ips.FirstOrDefault()?.ToString();
+            hostWrapper = new OrleansHostWrapper(config);
+            return hostWrapper.Run();
+        }
+
+        private static int ShutdownSilo()
+        {
+            if (hostWrapper != null)
+            {
+                return hostWrapper.Stop();
+            }
+            return 0;
+        }
+    }
+}
+```
+
+`Program.cs`(client):
+
+```csharp
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Runtime.Configuration;
+using OrleansGrainInterfaces;
+
+namespace OrleansClient
+{
+    class Program
+    {
+        private static IClusterClient client;
+        private static bool running;
+
+        static void Main(string[] args)
+        {
+            Task.Run(() => InitializeOrleans());
+
+            Console.ReadLine();
+
+            running = false;
+        }
+
+        static async Task InitializeOrleans()
+        {
+            var config = new ClientConfiguration();
+            config.DeploymentId = "Orleans-Docker";
+            config.PropagateActivityId = true;
+            var hostEntry = await Dns.GetHostEntryAsync("orleans-silo");
```

Comment:
When I use service name "orleans-silo", have a error.

Unhandled Exception: System.AggregateException: One or more errors occurred. (No such device or address)

It's my docker-compose file:
```
version: '3'

services:
  orleans-silo:
    image: hub.xxx.com/beck/orleans/silo-test:0.0.1
    volumes:
      - /usr/local/beck/docker/orleans/test/siloservice/config:/app/Config
~                                                                             
```